pkg/controller/machineset/controller.go

```diff
@@ -204,5 +247,22 @@ func (c *MachineSetControllerImpl) adoptOrphan(machineSet *v1alpha1.MachineSet,
 	machine.ObjectMeta.SetOwnerReferences(ownerRefs)
 	if _, err := c.clusterAPIClient.ClusterV1alpha1().Machines(machineSet.Namespace).Update(machine); err != nil {
 		glog.Warningf("Failed to update machine owner reference. %v", err)
+		return err
 	}
+	return nil
+}
+
+func (c *MachineSetControllerImpl) enqueueAfter(machineSet *v1alpha1.MachineSet, after time.Duration) {
+	key, err := cache.MetaNamespaceKeyFunc(machineSet)
+	if err != nil {
+		utilruntime.HandleError(fmt.Errorf("couldn't get key for object %+v: %v", machineSet, after))
+		return
+	}
+	c.queue.AddAfter(key, after)
```

Comment:
When do items leave this queue? It looks like it is growing unbounded.

Reply:
The queue is processed by the underlying controller library.
The queue has always been there, we are just adding items to it.

Reply:
I'm confused by this comment as it appears the queue is being defined on line 60 and is new as part of this PR?

Reply:
my apologies, i was missing some code in the init to have it point to the queue from the library.
thanks for catching that.

Reply:
Can you point me to the documentation / underlying library so I can understand how this processing will work?

Reply:
So I added this line, https://github.com/kubernetes-sigs/cluster-api/pull/193/files#diff-b0f93c81bd86491a95181a762269c619R76 to reference the underlying queue

you can see the underlying use of the queue here
https://github.com/kubernetes-sigs/cluster-api/blob/master/vendor/github.com/kubernetes-incubator/apiserver-builder/pkg/controller/controller.go#L32

and the set up of the queue here
https://github.com/kubernetes-sigs/cluster-api/blob/master/pkg/controller/machineset/zz_generated.api.register.go#L48